### PR TITLE
New Prettier

### PR DESCRIPTION
**Pull request by bh2smith**
Since the contracts have changed so much since the first attempt #228 - we have started over from the current master branch. Note that I think these changes make the contract munch worse and (at least at this stage) I would prefer not to incorporate these changes.

Note that `submitSolution` has gone from being < 57 lines to 113.

Had to downgrade the following solhint errors to warning:

- "function-max-lines"
- "two-lines-top-level-separator"
- "separate-by-one-line-in-contract": "warn",
- "bracket-align": "warn",

see `solhint.json` for other changes.

I would argue that most of the change made to the contract actually make it more difficult to read

### CODE REVIEW

**Comment by josojo** [comment]
Hey, thanks for re-preparing it!

I am very BOLD in using it. I configured vscode that the solidity code is autoformatted on every save of a .sol file. This is so powerful, it saves so much time. Love it!

Yeah, some formatting is questionable, but the autoformatting is such an awesome feature, that I want it to be merged

**Review by josojo — APPROVED** [review]
YES YES YES

File: contracts/EpochTokenLocker.sol [hunk 1]
```diff
@@ -64,10 +59,22 @@ contract EpochTokenLocker {
...
+            msg.sender,
+            address(this),
+            amount
+        );
         // solhint-disable-next-line max-line-length
-        balanceStates[msg.sender][token].pendingDeposits.amount = balanceStates[msg.sender][token].pendingDeposits.amount.add(amount);
-        balanceStates[msg.sender][token].pendingDeposits.stateIndex = getCurrentBatchId();
+        balanceStates[msg.sender][token]
+            .pendingDeposits
+            .amount = balanceStates[msg.sender][token]
```
Comment (fleupold) [inline contracts/EpochTokenLocker.sol:71]: This formatting is very awkward as it reads like one long call chain (but wait, there is an assignment in the middle). At least the rhs callchain should be indented more in my opinion.

File: contracts/SnappAuction.sol [hunk 2]
```diff
@@ -3,49 +3,48 @@ pragma solidity ^0.5.0;
 import "./SnappBase.sol";
 import "solidity-bytes-utils/contracts/BytesLib.sol";
 
-
 contract SnappAuction is SnappBase {
     using BytesLib for bytes;
 
-    uint public constant MAX_UINT = 2**256 - 1;
+    uint256 public constant MAX_UINT = 2**256 - 1;
```
Comment (fleupold) [inline contracts/SnappAuction.sol:9]: I love this rule!

File: .solhint.json [hunk 3]
```diff
@@ -23,6 +23,10 @@
...
         "no-simple-event-func-name": true,
-        "not-rely-on-time": false
+        "not-rely-on-time": false,
+        "no-empty-blocks": "off",
+        "two-lines-top-level-separator": "warn",
+        "separate-by-one-line-in-contract": "warn",
+        "bracket-align": "warn"
     }
-}
+}
```
Comment (nlordell) [inline .solhint.json:32]: And the war of new line at EOF continues....

File: .soliumrc.json [hunk 4]
```diff
@@ -12,10 +12,6 @@
       "error",
       "double"
     ],
-    "indentation": [
-      "error",
-      4
-    ],
```
Comment (nlordell) [inline .soliumrc.json:18]: Is this because of pretty formatting?

File: contracts/DevDependencies.sol [hunk 5]
```diff
@@ -9,6 +9,4 @@ import "openzeppelin-solidity/contracts/token/ERC20/ERC20Mintable.sol";
 import "solidity-multicall/contracts/MultiCaller.sol";
 import "@gnosis.pm/owl-token/contracts/TokenOWLProxy.sol";
 
-
-contract DevDependencies { // solhint-disable no-empty-blocks
-}
\ No newline at end of file
+contract DevDependencies {}
```
Comment (nlordell) [inline contracts/DevDependencies.sol:12]: :+1:

File: package.json [hunk 6]
```diff
@@ -16,7 +16,8 @@
     "networks-reset": "mkdir -p build/contracts && npx truffle networks --clean && npm run networks-inject",
     "verify-stablex": "truffle run verify StablecoinConverter",
     "lint": "./node_modules/.bin/eslint .",
-    "solhint": "./node_modules/.bin/solhint \"contracts/**/*.sol\""
+    "solhint": "./node_modules/.bin/solhint \"contracts/**/*.sol\"",
+    "prettier": "./node_modules/.bin/prettier --write 'contracts/**/*.sol'"
```
Comment (nlordell) [inline package.json]:
```suggestion
    "prettier": "prettier --write 'contracts/**/*.sol'"
```

**Review by nlordell — APPROVED** [review]
The code is so much more legible now its nuts! And the best part the computer does this for you.

I only have one additional request: maybe a `prettier --check` (if that is how it is done) can be run on CI to disallow unformatted code.

File: .soliumrc.json (hunk 4 above)
Comment (bh2smith) [reply .soliumrc.json:18]: yes sir it was.

File: .solhint.json (hunk 3 above)
Comment (bh2smith) [reply .solhint.json:32]: Another place where I don't like the auto-changes imposed by this

File: package.json (hunk 6 above)
Comment (bh2smith) [reply package.json]: If this works, then yes!

**Review by fleupold — APPROVED** [review]

File: contracts/EpochTokenLocker.sol (hunk 1 above)
Comment (bh2smith) [reply contracts/EpochTokenLocker.sol:71]: Its horrible I know...  at the moment I am not able to deal with it.

**Comment by bh2smith** [comment]
`pretty -check` has been include in Travis CI. Good luck with that!

**Pull request merged**